File: Starters/Starters 43/DNASTORAGE/Solution.c
#include <stdio.h>

int main() {
    int t,i,j,len;
    scanf("%d",&t);
    for(i=0;i<t;i++){
        scanf("%d",&len);
        char arr[len];
        scanf("%s",arr);
        for(j=0;arr[j]!='\0';j=j+2){
            if((arr[j] == '0') && (arr[j+1]== '0')) 
            printf("A");
        else if((arr[j] == '0') && (arr[j+1]== '1')) 
        printf("T");
        else if((arr[j] == '1') && (arr[j+1]== '0')) 
        printf("C");
        else if((arr[j] == '1') && (arr[j+1]== '1')) 
        printf("G");
        }
        printf("\n");
    }
    
	return 0;
}

File: Starters/Starters 43/DNASTORAGE/Solution.go
package main
import "fmt"

func main(){
    var arr string
    var lent,t int
    fmt.Scanln(&t)
    for i := 0; i<t; i++{
        fmt.Scanln(&lent)
        fmt.Scanln(&arr)
        for j :=0 ; j<lent ; j+=2{
            if arr[j] == '0' && arr[j+1] == '0' {
                fmt.Print("A")
              } else if arr[j] == '0' && arr[j+1] == '1' {
                fmt.Print("T")
              } else if arr[j] == '1' && arr[j+1] == '0' {
                fmt.Print("C")
              } else {
                fmt.Print("G")
              }
        }
          fmt.Println("")
    }
 }
